src/main/java/org/jabref/logic/importer/fetcher/CrossRef.java

```diff
@@ -24,96 +37,145 @@
  *
  * See https://github.com/CrossRef/rest-api-doc
  */
-public class CrossRef {
-    private static final Log LOGGER = LogFactory.getLog(CrossRef.class);
+public class CrossRef implements IdParserFetcher<DOI>, EntryBasedParserFetcher, SearchBasedParserFetcher, IdBasedParserFetcher {
```

Comment:
I mean you obviously worked a lot on this, so it is not easy to give a good suggestion. I'll try to take a look on this in the next days. If I am not able to come up with a good solution we will keep it the way it is.